### PR TITLE
[Debt] Refactor user table to use `ResponsiveTable`

### LINKED ISSUE #7692: ♻️ Refactor UserTable to use new table component
## ♻️ Debt/Refactor

We want to use new table component instead of old ApiManagedTable.

## 🕵️ Details

Add any additional details that could assist with the development of the new feature.

## 🙋‍♀️ Proposed Solution

(optional) Do you have a proposed solution?

## 🌎 Localization

n/a

## ✅ Acceptance Criteria

- [ ] UserTable.tsx uses new table component, does not reference ApiManagedTable, but functionality remains the same

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #7692 

## 👋 Introduction

Refactors the existing users table to make use of the new `ResponsiveTable` component.

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

1. Build `npm run dev`
2. Navigate to `/admin/users`
3. Interact with the table to confirm it works as expected

## 📸 Screenshot
localhost_8000_en_admin_users

### CODE REVIEW

File: apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx [hunk 1]
```diff
@@ -246,31 +247,26 @@ const Actions = ({
...
+                      id: "mxOuYK",
+                      description:
+                        "Text label for button to download a csv file of items in a table.",
+                    })}
+                </DownloadCsv>
               </span>
             )}
 
-            {print?.onPrint && (
+            {(print?.onPrint || print?.component) && (
```
Comment (brindasasi) [inline apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx:266]: Does this work for you? I get error as `Property 'component' does not exist on type 'DatasetPrint`

Comment (esizer) [reply apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx:266]: Hmm, it does seem to work for me and it does exist on that type:

https://github.com/GCTC-NTGC/gc-digital-talent/blob/8f3366d60088aff1edbfb128e29e818fdd73a48b/apps/web/src/components/Table/ResponsiveTable/types.ts#L109-L113

And, it seems to function 🤔 
Screenshot 2023-11-08 103639

**Comment by brindasasi** [comment]
There is a small issue with pagination .. if I select something on the first page and navigate to different page , the selection changes on every page. I mean if I select last 2 records of first page and go to second page, last 2 records of second page has been selected. 
https://github.com/GCTC-NTGC/gc-digital-talent/assets/89482910/ea449042-63eb-4ece-b851-b2d91b456bdf

**Comment by esizer** [comment]
> There is a small issue with pagination .. if I select something on the first page and navigate to different page , the selection changes on every page. I mean if I select last 2 records of first page and go to second page, last 2 records of second page has been selected.

ooo, nice find! It should be fixed in [reset row selection on data change](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8416/commits/256fed7b9bf6fea34ba47c455e9e8e4af3694336)

File: apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx
Comment (vd1992) [inline apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx:1]: Something appears to have gone wrong here with the selection

https://recordit.co/TUYKdGm4QS

File: apps/web/src/pages/Users/IndexUserPage/components/UserTable.tsx
Comment (vd1992) [inline apps/web/src/pages/Users/IndexUserPage/components/UserTable.tsx:1]: Probably not applicable to just this table but should the bottom bar not extend all the way to match the top? You need to add multiple columns to force the overflow before this becomes apparent

image

**Review by vd1992 — CHANGES_REQUESTED** [review]
Quick look-through

File: apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx
Comment (vd1992) [reply apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx:1]: Same thing is apparent for the CSV file too if you download

Comment (esizer) [reply apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx:1]: Okay! I fixed this, turns out...The table was aware of the IDs we selected but the callback function was incorrectly sending the indices still causing an odd issue. Also, while I was working on that I noticed the pagincation from seach params was odd by one so fixed that too 😅 

[fix row selection, initial state from params](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8416/commits/95d830946c40a7aba9e230bcc551428fe17deed8)

File: apps/web/src/pages/Users/IndexUserPage/components/UserTable.tsx
Comment (esizer) [reply apps/web/src/pages/Users/IndexUserPage/components/UserTable.tsx:1]: And, made the footer sticky to the left in [fix table actions to left](https://github.com/GCTC-NTGC/gc-digital-talent/pull/8416/commits/6ff4adbcc6680c1a35ce86588d4c4c8c4f22a124)

File: apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx
Comment (vd1992) [reply apps/web/src/components/Table/ResponsiveTable/RowSelection.tsx:1]: Well that looked confusing 😵

**Review by vd1992 — APPROVED** [review]
I didn't do too thorough a dive, but putting the table through its paces and I can't find anything breaking. I think Brinda is still reviewing so please wait for her

Otherwise, great! Impressive table so it's nice to see it in more places.

**Comment by brindasasi** [comment]
on it!

**Review by brindasasi — APPROVED** [review]
Pagination works very well. Looking very cool! thanks! 🥳

**Pull request merged**